wordpress-mobile/WordPress-iOS
# Stats: Reuse Stats Store instances between VCs and VMs to share the state

ViewModel may load the Store state that VC tries to access. If the Stores have different instances, then the state will be different.

I noticed the issue with `Post Stats` page. `PostStatsViewModel` fetches post information and then notifies VC about the change. VC accesses Store state to get available dates for the header. Since VM and VC uses different Stores, VC store doesn't have any state and therefore the header doesn't allow to switch between dates.

A short-term solution is to reuse Store instances between VC and VM. 

Potentially a better solution would be not to access store at all in VCs, and get all the information from View Models.

### To test:

1. Open Stats
2. Day
3. Post & Pages
4. Select a post
5. Confirm that the header allows to switch between dates

#### Before

https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/5301d79c-1306-4493-9685-aa576b7723d6

#### After

https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/6668a295-648a-42cc-b8d9-d9440359c931


## Regression Notes
1. Potential unintended areas of impact

None, it also avoids any other potential undetected issues.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:


Testing checklist:
